viadot/tasks/customer_gauge.py

```diff
@@ -0,0 +1,146 @@
+
+        Args:
+            endpoint (Literal["responses", "non-responses"]): Indicate which endpoint to connect. Defaults to None.
+            total_load (bool, optional): Indicate whether to download the data to the latest. If 'False', only one API call is executed (up to 1000 records). Defaults to True.
+            cursor (int, optional): Cursor value to navigate to the page. Defaults to None.
+            pagesize (int, optional): Number of responses (records) returned per page, max value = 1000. Defaults to 1000.
+            date_field (Literal["date_creation", "date_order", "date_sent", "date_survey_response"], optional): Specifies the date type which filter date range. Defaults to None.
+            start_date (datetime, optional): Defines the period end date in yyyy-mm-dd format. Defaults to None.
+            end_date (datetime, optional): Defines the period start date in yyyy-mm-dd format. Defaults to None.
+            credentials_secret (str, optional): The name of the Azure Key Vault secret containing a dictionary with ['client_id', 'client_secret']. Defaults to "CUSTOMER-GAUGE".
+            vault_name (str, optional): The name of the vault from which to obtain the secret. Defaults to None.
+
+        Returns:
+            pd.DataFrame: Final pandas DataFrame.
+        """
+        try:
+            credentials_str = AzureKeyVaultSecret(
+                credentials_secret, vault_name=vault_name
+            ).run()
+        except ValueError:
+            pass
+
+        credentials = json.loads(credentials_str)
+
+        df_list = []
+
+        customer_gauge = CustomerGauge(endpoint=endpoint, credentials=credentials)
+        logger.info(f"Starting downloading data from {self.endpoint} endpoint...")
+        json_data = customer_gauge.get_json_response(
+            cursor=cursor,
+            pagesize=pagesize,
+            date_field=date_field,
+            start_date=start_date,
+            end_date=end_date,
+        )
+        cur = customer_gauge.get_cursor(json_data)
+        df = customer_gauge.to_df(json_data)
+        df_list.append(df)
+        if total_load == True:
+            if cursor is None:
+                logger.info(
+                    f"Downloading all the data from the {self.endpoint} endpoint. Process might take a few minutes..."
+                )
+            else:
+                logger.info(
+                    f"Downloading starting from the {cursor} cursor. Process might take a few minutes..."
+                )
+            while df.empty == False:
+                json_data = customer_gauge.get_json_response(cursor=cur)
+                cur = customer_gauge.get_cursor(json_data)
+                df = customer_gauge.to_df(json_data)
+                df_list.append(df)
+
+        df_total = pd.concat(df_list, ignore_index=True)
+        logger.info(f"Downloading finished.")
```

Comment:
I am not sure if this logger is necessary here because at the end after the task is finished in logger, we can see the message that `Finished task run for task with final state: 'Success'`

Reply:
right, now it looks like that:
![image](https://user-images.githubusercontent.com/84125127/230124224-1ab2c9f1-45d0-4de4-9431-c906f8d145f4.png)


I will remove this one